### PR TITLE
Add sample for aggregator

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

### CODE REVIEW

File: aggregator/aggregator.bal [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import ballerina/http;
+
+map<json[]> survayMap = {};
```
Comment (manuranga) [inline aggregator/aggregator.bal]: survayData or surveys or partialSurveys

File: aggregator/aggregator.bal [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        if survayMap.hasKey(userId) {
+            survayData.push(...survayMap.get(userId));
+        }
+        survayData.push(formData);
+        if survayData.length() == 3 {
+            json _ = check formSubmitClient->/survay/[id]/submit.post({userId: survayData});
+            _ = survayMap.remove(userId);
+        } else {
+            survayMap[userId] = survayData;
+        }
```
Comment (manuranga) [inline aggregator/aggregator.bal]: This looks up map twice. Eg fix:
```ballerina
  PartialSurvey? existingSurvay = partialSurveys[userId];
  if existingSurvay == () {
      PartialSurvey newSurvay = [data];
      partialSurveys[userId] = newSurvay;
  } else {
      existingSurvay.push(data);
      if survayData.length() == 3 {
          _ = check formSubmitClient->/survay/[id]/submit.post({userId: survayData});
          _ = survayMap.remove(userId);
      }
  }
```

Comment (LakshanWeerasinghe) [reply aggregator/aggregator.bal]: updated.

File: aggregator/aggregator.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+tagline: Wait for a sequence of messages and combine them
+desc: Aggregator patiently collects a sequence of messages and combine them once all have been received.
+helps: Ballerina provides convenient map and table data structures for temporary message storage.
```
Comment (manuranga) [inline aggregator/aggregator.yml]: Mention we also have libraries for Redis for distributed storage and database support for permanent storage.

Comment (manuranga) [reply aggregator/aggregator.yml]: also mention our cache library, when users need control of automatic clean up.

File: aggregator/aggregator.bal [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+final http:Client formSubmitClient = check new ("http://api.survayme.com.balmock.io");
+
+service /api/v1 on new http:Listener(8080) {
+
+    resource function post survay/[string id](@http:Header string userId, @http:Payload json formData) returns error? {
+        json[]? survayData = partialSurveys[userId];
+        if survayData == () {
+            json[] newSurvay = [formData];
+            partialSurveys[userId] = newSurvay;
```
Comment (SasinduDilshara) [inline aggregator/aggregator.bal:13]:
```suggestion
            partialSurveys[userId] = [formData];
```

**Review by manuranga — APPROVED** [review]

**Pull request merged**